InsiteTeamTask/Controllers/AttendanceController.cs

```diff
@@ -1,23 +1,59 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using InsiteTeamTask.Models;
-using InsiteTeamTask.Repositories;
+using InsiteTeamTask.Services;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
 
 namespace InsiteTeamTask.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class AttendanceController : ControllerBase
     {
+        private readonly IAttendenceViewService _attendenceViewService;
+        private readonly ILogger<AttendanceController> _logger;
+
+        public AttendanceController(IAttendenceViewService attendenceViewService, ILogger<AttendanceController> logger)
+        {
+            _attendenceViewService = attendenceViewService;
+            _logger = logger;
+        }
+
         // GET api/values
-        [HttpGet]
-        public ActionResult<IEnumerable<Attendance>> Get()
+        [HttpGet()]
+        [Route("get/{gameNumber}/{seasonNumber}")]
+        public ActionResult<AttendenceViewModel> Get(int gameNumber, int seasonNumber)
         {
-            var repo = new DataRepository();
+            try
+            {
+                //Using a view service so the controller has minimal logic.
+                var viewModel = _attendenceViewService.GetAttendence(gameNumber, seasonNumber);
 
-            var attendance = repo.GetAttendanceListFor(gameNumber: 3);
+                return Ok(viewModel);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error while trying to retrieve game attendence");
+                return StatusCode(500);
```

Comment:
Hiding any error details from the client leaves the user confused and frustrated. Logging the error client-side is good, but you'd be better off throwing the exception here so that the client has something more to go on than "500".

If you fancied making this more intelligent, you could catch exceptions in middleware and then return some pre-specified error objects based on the specific exception thrown.